### PR TITLE
Integration test for Databricks delta_lake mode

### PR DESCRIPTION
**Pull request by Sevenannn**
# 🗣 Description

Add integration test for Databricks Delta_lake mode

## ⛓️‍💥 Breaking Change

* [ ] "Breaking Change" label added if this PR is a breaking change

### CODE REVIEW

File: crates/runtime/tests/databricks_delta/mod.rs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+                    // snapshot the values of the results
+                    let results = arrow::util::pretty::pretty_format_batches(&result_batches)
+                        .expect("should pretty print result batch");
+                    insta::with_settings!({
+                        description => format!("File Integration Test Results"),
+                        omit_expression => true,
+                        snapshot_path => "../snapshots"
+                    }, {
+                        insta::assert_snapshot!(format!("file_integration_test_select"), results);
```
Comment (sgrebnov) [inline crates/runtime/tests/databricks_delta/mod.rs]:
```suggestion
                        insta::assert_snapshot!(format!("databricks_delta_lake_select"), results);
```

File: crates/runtime/tests/databricks_delta/mod.rs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                Some(Box::new(|result_batches| {
+                    for batch in &result_batches {
+                        assert_eq!(batch.num_columns(), 15, "num_cols: {}", batch.num_columns());
+                        assert_eq!(batch.num_rows(), 1, "num_rows: {}", batch.num_rows());
+                    }
+
+                    // snapshot the values of the results
+                    let results = arrow::util::pretty::pretty_format_batches(&result_batches)
+                        .expect("should pretty print result batch");
+                    insta::with_settings!({
```
Comment (sgrebnov) [inline crates/runtime/tests/databricks_delta/mod.rs:104]: Please review if `insta::with_settings!` is  required vs we can avoid this for simplicity

File: crates/runtime/tests/databricks_delta/mod.rs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                    }, {
+                        insta::assert_snapshot!(format!("file_integration_test_select"), results);
+                    });
+                })),
+            )];
+
+            for (query, snapshot_suffix, validate_result) in queries {
+                run_query_and_check_results(
+                    &mut rt,
+                    &format!("file_integration_test_{snapshot_suffix}"),
```
Comment (sgrebnov) [inline crates/runtime/tests/databricks_delta/mod.rs]:
```suggestion
                    &format!("databricks_delta_lake_test_{snapshot_suffix}"),
```

File: crates/runtime/tests/databricks_delta/mod.rs (hunk 2 above)
Comment (Sevenannn) [reply crates/runtime/tests/databricks_delta/mod.rs:104]: It's needed for snapshots to be placed in the right directory

**Review by sgrebnov — APPROVED** [review]

**Pull request merged**